src/Currency.php

```diff
@@ -144,16 +145,30 @@ public function getAmountAsFractionalUnits(): int
     }
 
     /**
+     * @todo Use method getDecorationType() when decorating the result
      * {@inheritdoc}
      */
-    public function format(string $decimalsSeparator = '.', string $thousandsSeparator = '', int $extraPrecision = 0): string
+    public function format(CurrencyFormatInterface $currencyFormat = null): string
     {
-        $nDecimals = $this->currencyType->getNumFractionalDigits() + $extraPrecision;
+        if (is_null($currencyFormat)) {
+            $currencyFormat = new CurrencyFormat();
+        }
+
+        $nDecimals = $currencyFormat->getPrecision();
+        if (is_null($nDecimals)) {
+            $nDecimals = $this->currencyType->getNumFractionalDigits() + $currencyFormat->getExtraPrecision();
+        }
+
         $amount = Decimal::fromDecimal($this->amount, $nDecimals);
 
-        $number = ('' === $thousandsSeparator)
-            ? \str_replace('.', $decimalsSeparator, $amount->__toString())  // This is safer!
-            : \number_format($amount->asFloat(), $nDecimals, $decimalsSeparator, $thousandsSeparator);
+        $number = ('' === $currencyFormat->getThousandsSeparator())
+            ? \str_replace('.', $currencyFormat->getDecimalsSeparator(), $amount->__toString())  // This is safer!
+            : \number_format(
+                $amount->asFloat(),
+                $nDecimals,
+                $currencyFormat->getDecimalsSeparator(),
+                $currencyFormat->getThousandsSeparator()
+            );
 
         return ($this->currencyType->getSymbolPlacement() === CurrencyType::BEFORE_PLACEMENT)
```

Comment:
Here we can use the `$currencyFormat->getDecorationType` call to know if we have to use the currency symbol, the currency ISO code, or no decoration at all.

Remember that the "symbol placement" is only valid for symbols, ISO codes are always placed after the numeric amount.